docs/devcontainer-spec-support.md

```diff
@@ -0,0 +1,72 @@
+| 🔴                            | `userEnvProbe`                        | Shell to use when probing for user environment variables.                                                              | -                                                      |
+| 🔴                            | `overrideCommand`                     | Override the default sleep command to be run by supporting services.                                                   | -                                                      |
+| 🔴                            | `shutdownAction`                      | Action to take when supporting tools are closed or shut down.                                                          | -                                                      |
+| 🔴                            | `init`                                | Adds a tiny init process to the container.                                                                             | [#221](https://github.com/coder/envbuilder/issues/221) |
+| 🔴                            | `privileged`                          | Whether the container should be run in privileged mode.                                                                | -                                                      |
+| 🔴                            | `capAdd`                              | Capabilities to add to the container (for example, `SYS_PTRACE`).                                                      | -                                                      |
+| 🔴                            | `securityOpt`                         | Security options to add to the container (for example, `seccomp=unconfined`).                                          | -                                                      |
+| 🔴                            | `mounts`                              | Add additional mounts to the container.                                                                                | [#220](https://github.com/coder/envbuilder/issues/220) |
+| 🟢                            | `features`                            | Features to be added to the devcontainer.                                                                              | -                                                      |
+| 🔴                            | `overrideFeatureInstallOrder`         | Override the order in which features should be installed.                                                              | [#226](https://github.com/coder/envbuilder/issues/226) |
+| 🔴                            | `customizations`                      | Product-specific properties, e.g., _VS Code_ settings and extensions.                                                  | [#43](https://github.com/coder/envbuilder/issues/43)   |
```

Comment:
I'm not sure red/unsupported is right here. I see this more as a responsibility of the tool that's being customized to apply them to itself.

Maybe a new category like gray = not envbuilders responsibility. 😅

Reply:
🤔 If it's not envbuilder's responsibility, why is there an open issue to support VSCode-specific customizations?

Reply:
Maybe we should move this doc into `coder/coder` then. Auto-installing VS Code Extensions via `devcontainer.json` will be a top feature request for Coder and I think its important for us to communicate if this use case is possible or not.

Maybe we should say "partially supported" and link to the coder/coder issue and workaround?

Reply:
@bpmct which specific `coder/coder` issue should we link to? The workaround for installing features is documented in envbuilder#43.

Reply:
> Maybe we should move this doc into coder/coder then. 

I really don't think this document belongs in `coder/coder`. It's specific to `coder/envbuilder`.

Reply:
Gotcha. Then let's link to the envbuilder issue/workaround.

Reply:
I don't think coder/coder is the best place either. Just to clarify my POV:

My preferred way to support VS Code customizations would be to "trick" VS Code into believing it is inside a Dev Container. This would make VS Code automatically apply the customizations from the `devcontainer.json`.

Now, should the "tricking" part be a part of envbuilder? Maybe. Or maybe it should be part of the Coder VS Code extension. Or maybe the dev container spec needs to be expanded (via proposal) so that there's an official way to define it. The tool (VS Code) already supports installing extensions, applying settings, the issue is that we're not triggering that mode inside an envbuilder container.

I would like for us to think about the issue of customizations in this manner, **can we get `foobar` to do it by itself?**. Anything outside that scope is IMO the responsibility of `foobar`, a coder module, editor extension, etc.